uxarray/grid.py

```diff
@@ -141,6 +151,60 @@ def __init_grid_var_attrs__(self) -> None:
                 if value in self.ds.dims:
                     setattr(self, key, len(self.ds[value]))
 
+    def __from_multi_faces_vert__(self, dataset):
+        """Create a grid with multiple faces with vertices specified by the
+        given argument. Called by :func:`__init__`.
+
+        Parameters
+        ----------
+        dataset : ndarray
+            ndarry of vertex coordinates that form multiple faces.
+        """
+        self.ds = xr.Dataset()
+        self.ds["Mesh2"] = xr.DataArray(
+            attrs={
+                "cf_role": "mesh_topology",
+                "long_name": "Topology data of unstructured mesh",
+                "topology_dimension": -1,
+                "node_coordinates": "Mesh2_node_x Mesh2_node_y Mesh2_node_z",
+                "node_dimension": "nMesh2_node",
+                "face_node_connectivity": "Mesh2_face_nodes",
+                "face_dimension": "nMesh2_face"
+            })
+        self.ds.Mesh2.attrs['topology_dimension'] = dataset.ndim
```

Comment:
@philipc2 
`dataset.ndim` will only work for `ndarray` here, I wonder if you can help me expand the type of `dataset` from `ndarray` to [xarray.Dataset](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.html#xarray.Dataset), ndarray, [list](https://docs.python.org/3/library/stdtypes.html#list), [tuple](https://docs.python.org/3/library/stdtypes.html#tuple), as specified in our API here? Thanks!

Reply:
See my fix below, should also address this